src/site/content/en/blog/keyboard-lock/index.md

```diff
@@ -0,0 +1,113 @@
+
+The [`Keyboard` interface](https://developer.mozilla.org/en-US/docs/Web/API/Keyboard) of the the Keyboard API provides functions that toggle capturing of key presses from the physical keyboard as well as getting information about the used [keyboard layout](https://developer.mozilla.org/en-US/docs/Web/API/Keyboard/getLayoutMap).
+
+### Prerequisite
+
+There are two different types of full screen available in modern browsers: JavaScript-initiated via the [Fullscreen API](https://developer.mozilla.org/en-US/docs/Web/API/Fullscreen_API) and user-initiated via a keyboard shortcut. The Keyboard Lock API is only available when [JavaScript-initiated full screen](https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullscreen) is active.
+Here's an example of JavaScript-initiated full screen:
+
+```js
+await document.documentElement.requestFullscreen();
+```
+
+### Browser compatibility
+
+You can see browser compatibility on [Can I use](https://caniuse.com/#feat=mdn-api_keyboard_lock). Note that not all system keys can be locked. This varies from operating system to operating system. For example, follow [crbug.com/855738](https://crbug.com/855738) for progress updates on system keyboard lock for macOS.
+
+### Locking the keyboard
+
+The [`lock()`](https://developer.mozilla.org/en-US/docs/Web/API/Keyboard/lock) method of the `Keyboard` interface returns a promise after enabling the capture of key presses for any or all of the keys on the physical keyboard. This method can only capture keys that are granted access by the underlying operating system. The `lock()` method takes an array of one or more key codes to lock. If no key codes are provided, all keys will be locked. A list of valid key code values is available in the [UI Events KeyboardEvent code Values](https://www.w3.org/TR/uievents-code/#keyboard-key-codes) spec.
+
+#### Capturing all keys
+
+The following example captures all key presses.
+
+```js
+navigator.keyboard.lock();
+```
+
+#### Capturing specific keys
+
+The following example captures the <kbd>W</kbd>, <kbd>A</kbd>, <kbd>S</kbd>, and <kbd>D</kbd> keys. It captures these keys regardless of which modifiers are used with the key press. Assuming a US QWERTY layout, registering `"KeyW"` ensures that <kbd>W</kbd>, <kbd>Shift</kbd>&nbsp;+&nbsp;<kbd>W</kbd>, <kbd>Control</kbd>&nbsp;+&nbsp;<kbd>W</kbd>, <kbd>Control</kbd>&nbsp;+&nbsp;<kbd>Shift</kbd>&nbsp;+&nbsp;<kbd>W</kbd>, and all other key modifier combinations with <kbd>W</kbd> are sent to the app. The same applies to `"KeyA"`, `"KeyS"`, and `"KeyD"`.
+
+```js
+await navigator.keyboard.lock([
+  "KeyW",
+  "KeyA",
+  "KeyS",
+  "KeyD",
+]);
+```
+
+If `lock()` is called multiple times without an intervening call to `unlock()` (see [below](#unlocking-the-keyboard)), then only the key codes specified in the last call will be in effect. If a second call to `lock()` is made before the first one has finished, then the first one will be rejected with an `AbortError`.
```

Comment:
```suggestion
You can react on captured key presses for example in the `onkeydown` event:

```js
document.addEventListener('keydown', (e) => {
  if ((e.code === 'KeyA') && !(event.ctrlKey || event.metaKey)) {
    // Do something when the 'A' key was pressed, but only
    // when not in combination with the command or control key.
  }
});
```

If `lock()` is called multiple times without an intervening call to `unlock()` (see [below](#unlocking-the-keyboard)), then only the key codes specified in the last call will be in effect. If a second call to `lock()` is made before the first one has finished, then the first one will be rejected with an `AbortError`.